Version 1.2.1/os/os/.programs

```diff
@@ -76,11 +76,11 @@ term.clear()
  
 --Conditions
 if nOption  == 1 then
-shell.run("CactusOSWord")
+os.run({},"CactusOSWord")
 elseif nOption == 2 then
-shell.run("os/sendfile")
+os.run({},"os/sendfile")
 elseif nOption == 3 then
-shell.run("os/getfile")
+os.run({},"os/getfile")
 else
-shell.run(".menu")
-end
\ No newline at end of file
+os.run({},".menu")

```

Comment:
If you are running programs shell.run() might be the better option, because else the executed program will not have the ability to use the shell api, which might be required if you want to use programs made by others.

Reply:
Shell is CraftOS. If you need another OS for your OS, stop making the os.

Reply:
See [this](http://www.computercraft.info/forums2/index.php?/topic/29085-shouldnt-an-os-replace-the-dir-folder/page__view__findpost__p__271018)

Reply:
Compatibility does not mean leaving everything as is. You can implement shell if you need programs to use shell. Shell API should really never be used.

Reply:
I don't see any advantages in doing it like this.
For the one line the time save is minuscule and you basically break compatibility with a lot of already existing CC programs.

Reply:
It loads it without all of the environmental bs that shell runs it with. Less overhead, and less in the shells program stack, making the entire system run overall faster.